src/main/java/com/exactpro/th2/hand/services/MessageHandler.java

```diff
@@ -18,39 +18,102 @@
 
 import com.exactpro.remotehand.Configuration;
 import com.exactpro.remotehand.rhdata.RhScriptResult;
+import com.exactpro.th2.act.grpc.hand.ResultDetails;
 import com.exactpro.th2.act.grpc.hand.RhAction;
 import com.exactpro.th2.act.grpc.hand.RhActionsList;
+import com.exactpro.th2.act.grpc.hand.RhBatchResponse;
 import com.exactpro.th2.common.grpc.*;
+import com.exactpro.th2.common.schema.factory.CommonFactory;
 import com.exactpro.th2.hand.Config;
+import com.exactpro.th2.hand.RhConnectionManager;
 import com.exactpro.th2.hand.messages.RhResponseMessageBody;
+import com.exactpro.th2.hand.messages.eventpayload.EventPayloadMessage;
+import com.exactpro.th2.hand.messages.eventpayload.EventPayloadTable;
+import com.exactpro.th2.hand.messages.responseexecutor.ActionsBatchExecutorResponse;
+import com.exactpro.th2.hand.requestexecutors.ActionsBatchExecutor;
+import com.exactpro.th2.hand.scriptbuilders.ScriptBuilder;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.protobuf.*;
+import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.time.Instant;
 import java.util.*;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
 
-public class MessageHandler{
-
+public class MessageHandler implements AutoCloseable {
```

Comment:
This class is quite large. Probably move some methods to another classes? Probably all related with estore